solidity/dashboard/src/components/NetworkStatus.jsx

```diff
@@ -1,24 +1,27 @@
-import React, { useContext } from "react"
-import { Web3Context } from "./WithWeb3Context"
+import React from "react"
+import { useWeb3Context } from "./WithWeb3Context"
+import * as Icons from "./Icons"
 
 export const NetworkStatus = () => {
-  const { networkType, provider, error, yourAddress } = useContext(Web3Context)
+  const { networkType, provider, error, yourAddress } = useWeb3Context()
+
+  let status = "disconnected"
+  if (error) {
+    status = "error"
+  } else if (yourAddress && !error && provider !== null) {
```

Comment:
This feels like something the web3 context should expose directly as an `isConnected` or something, though I'm not sure 🤔